File: normal-251-Flatten_2D_Vector/251-Flatten_2D_Vector.py
class Vector2D:

    def __init__(self, v: List[List[int]]):
        self.memberList = v
        self.point = [0,0]


    def next(self) -> int:
        if len(self.memberList) > 0:
            if self.point[0] < len(self.memberList):
                if self.point[1] < len(self.memberList[self.point[0]]):
                    temp = self.memberList[self.point[0]][self.point[1]]
                    self.point[1] += 1
                    return temp
                else:
                    self.point[1] = 0
                    self.point[0] += 1
                    return self.next()
            else:
                return None
        else:
            return None

    def hasNext(self) -> bool:
        local_point = self.point.copy()
        while(True):
            if len(self.memberList) > 0:
                if local_point[0] < len(self.memberList):
                    if local_point[1] < len(self.memberList[local_point[0]]):
                        return True
                    else:
                        local_point[1] = 0
                        local_point[0] += 1
                else:
                    return False
            else:
                return False



# Your Vector2D object will be instantiated and called as such:
# obj = Vector2D(v)
# param_1 = obj.next()
# param_2 = obj.hasNext()

File: normal-251-Flatten_2D_Vector/251-Flatten_2D_Vector.cpp
class Vector2D {
private:
    vector<int> point;
    vector<vector<int>> memberList;
public:
    Vector2D(vector<vector<int>>& v): point({0, 0}) {
        memberList = v;
    }
    
    int next() {
        if(memberList.size() > 0)
        {
            if(point[0] < memberList.size())
            {
                if(point[1] < memberList[point[0]].size())
                {
                    point[1]++;
                    return memberList[point[0]][point[1]-1];
                }
                else
                {
                    point[1] = 0;
                    point[0] += 1;
                    return this->next();
                }
            }
            else
            {
                return 0;
            }
        }
        else
        {
            return 0;
        }
    }
    
    bool hasNext() {
        vector<int> localPoint = point;
        while(true)
        {
            if(memberList.size() > 0)
            {
                if(localPoint[0] < memberList.size())
                {
                    if(localPoint[1] < memberList[localPoint[0]].size())
                    {
                        return true;
                    }
                    else
                    {
                        localPoint[1] = 0;
                        localPoint[0]++;
                    }
                }
                else
                {
                    return false;
                }
            }
            else
            {
                return false;
            }
        }
    }
};

/**
 * Your Vector2D object will be instantiated and called as such:
 * Vector2D* obj = new Vector2D(v);
 * int param_1 = obj->next();
 * bool param_2 = obj->hasNext();
 */